src/Microsoft.AspNet.OData.Shared/ODataQueryContextExtensions.cs

```diff
@@ -26,5 +27,25 @@ public static ODataQuerySettings UpdateQuerySettings(this ODataQueryContext cont
 
             return updatedSettings;
         }
+
+        public static SkipTokenHandler GetSkipTokenHandler(this ODataQueryContext context)
+        {
+            if (context == null || context.RequestContainer == null)
+            {
+                return DefaultSkipTokenHandler.Instance;
+            }
+
+            return context.RequestContainer.GetRequiredService<SkipTokenHandler>();
+        }
+
+        public static SkipTokenQueryValidator GetSkipTokenQueryValidator(this ODataQueryContext context)
```

Comment:
>GetSkipTokenQueryValidator [](start = 46, length = 26)

Do we need this extensibility point? I.e., do folks need to plug in their own custom SkipTokenQueryValidator through DI? Can people plug in validators for other skip tokens through DI?

Reply:
To maintain parity with other query option validators, I enabled DI for the validator but since the skiptoken value can be defined by the service, it makes sense to give them the ability to validate that as well. 

---
In reply to: [287395079](https://github.com/OData/WebApi/pull/1724#discussion_r287395079) [](ancestors = 287395079)